src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1308,6 +1309,11 @@ public void clear(FeatureTypeInfo info) {
                 resultCRS = nativeCRS;
             } else {
                 resultCRS = getCRS(info.getSRS());
+                // force remoting re-projection incase of WFS-NG only
+                if (hints != null)
+                    if (hints.get(ResourcePool.MAP_CRS) != null
+                            && info.getMetadata().get(FeatureTypeInfo.OTHER_SRS) != null)
+                        resultCRS = (CoordinateReferenceSystem) hints.get(ResourcePool.MAP_CRS);
             }
```

Comment:
setting the result CRS to Getmap/GetFeature`s spatial filter, if the CRS is one of the OTHER supported SRS of this featuretype coming from WFS-NG Datastore

**Note to self: Verify the datasrouce is WFS-NG with usedefaultsrs set to false**